backend/pkg/x509engines/x509engine.go

```diff
@@ -108,36 +113,39 @@ func (engine X509Engine) CreateSubordinateCA(aki string, caID string, parentCACe
 		pubKey = signer.Public().(*ecdsa.PublicKey)
 	}
 
-	parentSN := helpers.SerialNumberToString(parentCACertificate.SerialNumber)
-	parentCASigner, err := parentEngine.cryptoEngine.GetPrivateKeyByID(CryptoAssetLRI(CertificateAuthority, parentSN))
+	parentCASigner, err := parentEngine.cryptoEngine.GetPrivateKeyByID(aki)
 	if err != nil {
-		lCEngine.Errorf("could not get parent signer key '%s': %s", parentSN, err)
-		return nil, err
+		lCEngine.Errorf("could not get parent signer key '%s': %s", aki, err)
+		return "", nil, err
 	}
 
 	templateCA.IsCA = true
 	certificateBytes, err := x509.CreateCertificate(rand.Reader, templateCA, parentCACertificate, pubKey, parentCASigner)
 	if err != nil {
 		lCEngine.Errorf("could not sign subordinate CA: %s", err)
-		return nil, err
+		return "", nil, err
 	}
 
 	certificate, err := x509.ParseCertificate(certificateBytes)
 	if err != nil {
 		lCEngine.Errorf("could not parse subordinate CA: %s", err)
-		return nil, err
+		return "", nil, err
 	}
 
-	return certificate, nil
+	return ski, certificate, nil
 }
 
 func (engine X509Engine) SignCertificateRequest(caCertificate *x509.Certificate, csr *x509.CertificateRequest, expirationDate time.Time) (*x509.Certificate, error) {
 	lCEngine.Debugf("starting csr signing with CA [%s]", caCertificate.Subject.CommonName)
 	lCEngine.Debugf("csr cn is [%s]", csr.Subject.CommonName)
-	caSn := helpers.SerialNumberToString(caCertificate.SerialNumber)
+
+	aki, err := software.NewSoftwareCryptoEngine(lCEngine).EncodePKIXPublicKeyDigest(caCertificate.PublicKey)
```

Comment:
Can we simplify this by creating the SoftwareCryptoEngine when instatiating the X509Engine? 

We can transform this line in something like 
`engine.software.EncodePKIXPublicKeyDigest(caCertificate.PublicKey)`

Reply:
Sure